camel/toolkits/notion_toolkit.py

```diff
@@ -0,0 +1,237 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import os
+from typing import List, Optional, cast
+
+from camel.toolkits import FunctionTool
+from camel.toolkits.base import BaseToolkit
+
+
+def get_plain_text_from_rich_text(rich_text):
+    r"""Extracts plain text from a list of rich text elements.
+
+    Args:
+        rich_text: A list of dictionaries representing rich text elements.
+            Each dictionary should contain a key named "plain_text" with
+            the plain text content.
+
+    Returns:
+        A string containing the combined plain text from all elements,
+        joined together.
+    """
+    plain_texts = [element.get("plain_text", "") for element in rich_text]
+    return "".join(plain_texts)
+
+
+def get_media_source_text(block):
+    r"""Extracts the source URL and optional caption from a
+    Notion media block.
+
+    Args:
+      block: A dictionary representing a Notion media block.
+
+    Returns:
+      A string containing the source URL and caption (if available),
+      separated by a colon.
+    """
+    if block.get(block.get("type"), {}).get("external"):
+        source = block[block["type"]]["external"]["url"]
+    elif block.get(block.get("type"), {}).get("file"):
+        source = block[block["type"]]["file"]["url"]
+    elif block.get(block.get("type"), {}).get("url"):
+        source = block[block["type"]]["url"]
+    else:
+        source = "[Missing case for media block types]: " + block.type
+
+    if len(block.get(block.get("type"), {}).get("caption", [])) > 0:
+        caption = get_plain_text_from_rich_text(
+            block[block["type"]]["caption"]
+        )
+        return caption + ": " + source
+
+    return source
+
+
+class NotionToolkit(BaseToolkit):
+    r"""A toolkit for retrieving information from the user's notion pages.
+
```

Comment:
Can also add the `Args:` section for the `__init__()` here

Reply:
Add the args to the init doc, following the same style which other files follow.

Reply:
I think now we have two styles in our codebase and have not been unified, but as Guohao mentioned last time I think it's better to stick with the style that put `Args:` section under the docstring of the class def?

Reply:
I think that this is just a style thing. We should follow one standard way. I was following the Google Python guide(https://google.github.io/styleguide/pyguide.html#384-classes). 
However I think this discussion shouldn't block the pr submission. We could have more discussions in our eng group. Then we could decide if we want to update the whole codebase to follow one style.